### ISSUE TITLE
Big Sur 11.2 diskutil: command not found

### ISSUE THREAD

**Issue by ihsanturk** [NixOS/nix#4488]
Install script can't find diskutil because the diskutil not in `$PATH`

**Steps To Reproduce**

1. Install macOS Big Sur 11.2
2. Run the installation command:
```sh
curl -L https://nixos.org/nix/install | sh
```
3. See error:
```
...unpack/nix-2.3.10-x86_64-darwin/install: line 115: diskutil: command not found
-:1: parser error : Document is empty
```
---

I think the solution is to call `diskutil` by its full path: `/usr/sbin/diskutil`.

**Comment by abathur** [comment]
I'm a little skeptical about whether /usr/sbin has been intentionally removed from the default PATH, but nailing these paths down has already been on my mind.

In a naked attempt to get it pushed through the pipeline, I've added a commit to my existing PR #4289 that nails down instances of diskutil by absolute path.

I suspect there are other utilities in /usr/sbin/ used in the installer, so I'll probably go ahead and audit the rest of them later this evening.

**Comment by abathur** [comment]
I got 11.2 20D64 installed on a system overnight (had to fresh install 11.1 after the upgrade from 11.0 -> 11.1 broke, rejoin beta program, and upgrade to 11.2) and it looks like my stock PATH is `/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin`.

Unless maybe you did a direct no-upgrade fresh install of 11.2, it seems like something in your RC/profile files may be causing this?

In any case, I think it's still fine to nail these down.

**Comment by ihsanturk** [comment]
- I was using 11.1 and was able to run the utilities in the `/usr/sbin/` so there were no problems. 
- Then I upgraded to 11.2 and deleted my `/nix` to get started from scratch (maybe I've messed up with my config files in this step?). 
- Finally issue happened. I checked the $PATH, it was `/bin:/usr/bin` (or `/usr/bin:/bin` I don't recall correctly).

I am sorry, I should have been written this in the issue description. I guess you are right about having my rc/profile files causing this.
